builtin/providers/azurerm/resource_arm_app_gateway.go

```diff
@@ -0,0 +1,1667 @@
+			ApplicationGatewayRequestRoutingRulePropertiesFormat: &network.ApplicationGatewayRequestRoutingRulePropertiesFormat{
+				RuleType: network.ApplicationGatewayRequestRoutingRuleType(ruleType),
+				HTTPListener: &network.SubResource{
+					ID: &httpListenerID,
+				},
+			},
+		}
+
+		if backendAddressPoolName := data["backend_address_pool_name"].(string); backendAddressPoolName != "" {
+			backendAddressPoolID := fmt.Sprintf("%s/backendAddressPools/%s", gatewayID, backendAddressPoolName)
+			rule.ApplicationGatewayRequestRoutingRulePropertiesFormat.BackendAddressPool = &network.SubResource{
+				ID: &backendAddressPoolID,
+			}
+		}
+
+		if backendHTTPSettingsName := data["backend_http_settings_name"].(string); backendHTTPSettingsName != "" {
+			backendHTTPSettingsID := fmt.Sprintf("%s/backendHttpSettingsCollection/%s", gatewayID, backendHTTPSettingsName)
+			rule.ApplicationGatewayRequestRoutingRulePropertiesFormat.BackendHTTPSettings = &network.SubResource{
+				ID: &backendHTTPSettingsID,
+			}
+		}
+
+		if urlPathMapName := data["url_path_map_name"].(string); urlPathMapName != "" {
+			urlPathMapID := fmt.Sprintf("%s/urlPathMaps/%s", gatewayID, urlPathMapName)
+			rule.ApplicationGatewayRequestRoutingRulePropertiesFormat.URLPathMap = &network.SubResource{
+				ID: &urlPathMapID,
+			}
+		}
+
+		rules = append(rules, rule)
+	}
+
+	return &rules
+}
+
+func expandAppGatewayURLPathMaps(d *schema.ResourceData, gatewayID string) *[]network.ApplicationGatewayURLPathMap {
+	configs := d.Get("url_path_map").([]interface{})
+	pathMaps := make([]network.ApplicationGatewayURLPathMap, 0, len(configs))
+
+	for _, configRaw := range configs {
+		data := configRaw.(map[string]interface{})
+
+		name := data["name"].(string)
+		defaultBackendAddressPoolName := data["default_backend_address_pool_name"].(string)
+		defaultBackendAddressPoolID := fmt.Sprintf("%s/backendAddressPools/%s", gatewayID, defaultBackendAddressPoolName)
+		defaultBackendHTTPSettingsName := data["default_backend_http_settings_name"].(string)
+		defaultBackendHTTPSettingsID := fmt.Sprintf("%s/backendHttpSettingsCollection/%s", gatewayID, defaultBackendHTTPSettingsName)
+
+		pathRules := []network.ApplicationGatewayPathRule{}
+		for _, ruleConfig := range data["path_rule"].([]interface{}) {
+			ruleConfigMap := ruleConfig.(map[string]interface{})
+
+			ruleName := ruleConfigMap["name"].(string)
+
+			rulePaths := []string{}
+			for _, rulePath := range ruleConfigMap["paths"].([]interface{}) {
+				rulePaths = append(rulePaths, rulePath.(string))
+			}
+
+			rule := network.ApplicationGatewayPathRule{
+				Name: &ruleName,
+				ApplicationGatewayPathRulePropertiesFormat: &network.ApplicationGatewayPathRulePropertiesFormat{
+					Paths: &rulePaths,
+				},
+			}
+
+			if backendAddressPoolName := ruleConfigMap["backend_address_pool_name"].(string); backendAddressPoolName != "" {
+				backendAddressPoolID := fmt.Sprintf("%s/backendAddressPools/%s", gatewayID, backendAddressPoolName)
```

Comment:
given there's no guarantee that these URL's won't change in the future, would it make more sense to pass in the Backend Address Pool ID, rather than the name?